integration/test/test_gpu_activity_agent.py

```diff
@@ -80,21 +81,42 @@ def launch_helper(experiment_type, experiment_args, app_conf, experiment_cli_arg
             phi_list=None,
         )
 
-        launch_helper(gpu_activity, experiment_args, parres.create_dgemm_appconf_cuda(mach, experiment_args), [])
+
+        path_nvidia = os.path.join(os.path.dirname(os.path.dirname(os.path.realpath(__file__))),
+                                                   "apps/parres/Kernels/Cxx11/dgemm-mpi-cublas")
+        path_intel = os.path.join(os.path.dirname(os.path.dirname(os.path.realpath(__file__))),
+                                                  "apps/parres/Kernels/Cxx11/dgemm-onemkl")
+        if os.path.exists(path_nvidia):
+            launch_helper(gpu_activity, experiment_args, parres.create_dgemm_appconf_cuda(mach, experiment_args), [])
+        elif os.path.exists(path_intel):
+            launch_helper(gpu_activity, experiment_args, parres.create_dgemm_appconf_oneapi(mach, experiment_args), [])
+        else:
+            self.fail("Neither NVIDIA or Intel dgemm variant was found")
```

Comment:
```suggestion
        if util.get_service_config_value('enable_nvml') == '1':
            app_path = os.path.join(os.path.dirname(os.path.dirname(os.path.realpath(__file__))),
                                                    "apps/parres/Kernels/Cxx11/dgemm-mpi-cublas")
            app_conf = parres.create_dgemm_appconf_cuda(mach, experiment_args)
        elif util.get_service_config_value('enable_levelzero') == '1':
            app_path = os.path.join(os.path.dirname(os.path.dirname(os.path.realpath(__file__))),
                                                    "apps/parres/Kernels/Cxx11/dgemm-onemkl")
            app_conf = parres.create_dgemm_appconf_oneapi(mach, experiment_args), [])
        if not os.path.exists(app_path):
            self.fail("Neither NVIDIA or Intel dgemm variant was found")
        launch_helper(gpu_activity, experiment_args, app_conf, [])
```

Please do a similar refactor to the stream case below.